test/UpdaterAndFilterEngineCreation.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-present eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <thread>
+#include <vector>
+#include <chrono>
+#include <string>
+#include <memory>
+
+#include "BaseJsTest.h"
+
+using namespace AdblockPlus;
+
+namespace
+{
+  class UpdaterAndFilterEngineCreationTest : public BaseJsTest
+  {
+  protected:
+
+    static const size_t COUNT = 100;
+    const std::string PROP_NAME = "patternsbackupinterval";
+    std::vector<std::thread> threadsVector;
```

Comment:
Basically, one can use `AdblockPlus::AsyncExecutor` from *include/AdblockPlus/AsyncExecutor.h*. It also should not be a member, just allocate it before the loop (below) and call `asyncExecutor.Dispatch(the-same-labda)`.

Reply:
OK, I'll check it and use AsyncExecutor.